### PR TITLE
[EXPLORER] Fix behavior from Show Desktop button

### PR DESCRIPTION
**Pull request by zecarlos1957**
## Purpose

Fix behavior from Show Desktop button.
JIRA issues: [CORE-19703](https://jira.reactos.org/browse/CORE-19703), [CORE-15369](https://jira.reactos.org/browse/CORE-15369)
Addendum to https://github.com/reactos/reactos/commit/70d7009ad958da5b0952b6e5af5194337a387459

## Observed behavior:

On the second click to "Show desktop" button, the minimized windows are not restored.

This happens because after the first click and the addition to `CSimpleArray<MINWNDPOS> g_MinimizedAll`, a call to `RestoreMinimizedNonTaskWnds()` via `HSHELL_WINDOWACTIVATED` message, emptied the list of minimized windows.

Don't empty `g_MinimizedAll` while processing HSHELL_WINDOWCREATED or HSHELL_WINDOWACTIVATED message. 
(bDestroyed=FALSE).
Empty  `g_MinimizedAll` only when SendPulseToTray is called with bDestroyed=TRUE.

### CODE REVIEW

**Comment by HBelusca** [comment]
Cc @Splitwirez too :)

**Comment by binarymaster** [comment]
Please link against [CORE-15369](https://jira.reactos.org/browse/CORE-15369) and mark as addendum to https://github.com/reactos/reactos/commit/70d7009ad958da5b0952b6e5af5194337a387459.

**Review by cbialorucki — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
Seems to be working fine. Fixes [CORE-19703](https://jira.reactos.org/browse/CORE-19703) and related ticket [CORE-18350](https://jira.reactos.org/browse/CORE-18350) has not regressed.

File: base/shell/explorer/traywnd.cpp [hunk 1]
```diff
@@ -3243,10 +3243,11 @@ class CTrayWindow :
...
         }
 
-        g_MinimizedAll.RemoveAll();
-
-        if (!bDestroyed)
+        if (bDestroyed)
+            g_MinimizedAll.RemoveAll();
+        else
             ::SetForegroundWindow(hwndActive);
+
```
Comment (binarymaster) [inline base/shell/explorer/traywnd.cpp:3250]:
```suggestion
```

**Pull request merged**